src/component.rs

```diff
@@ -113,12 +113,19 @@ pub fn register_component(world: *mut ecs_world_t, desc: ComponentDescriptor) ->
 	
 	// let s_id = 0;
 	let comp_desc = ecs_component_desc_t {
-        _canary: 0,
+		_canary: 0,
 		entity: e_desc,
+		#[cfg(all(target_arch = "wasm32", target_os = "emscripten"))]
```

Comment:
I wonder if we can replace all these with size_t instead, and avoid the cfg checks?